base_conditional_image_partner/models/conditional_image.py

```diff
@@ -0,0 +1,32 @@
+# Copyright 2022 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+from odoo import api, models
+
+
+class ConditionalImage(models.Model):
+    _inherit = "conditional.image"
+
+    @api.model
+    def create(self, val_list):
+        image = super().create(val_list)
+        image.reset_company_logo_web_for_partner()
+        return image
+
+    def write(self, vals):
+        result = super().write(vals)
+        self.reset_company_logo_web_for_partner()
+        return result
+
+    def reset_company_logo_web_for_partner(self):
+        """
+        Reset the company field logo_web for each related partner matching the image criteria
+        """
+        partner_model = self.env["res.partner"]
+        company_model = self.env["res.company"]
+        for record in self.filtered(lambda rec: rec.model_name == "res.partner"):
```

Comment:
Performance wise - as we've seen in our internal call - this is too heavy.
We should search for companies and partners the other way around.

Reply:
```suggestion
        for record in self.filtered(lambda rec: rec.model_name == "res.partner"):
        domain = []
        if record.company_id:
            domain = [("company_id", "=", record.company_id.id)]
        companies = company_model.search(domain)
        matching_companies = companies.filtered(
            lambda company: company.partner_id._conditional_image_evaluate_selector(record)
        )
        matching_companies._compute_logo_web()
```
@simahawk 
something like that ?